src/main/java/com/conveyal/gtfs/loader/JdbcTableWriter.java

```diff
@@ -0,0 +1,542 @@
+package com.conveyal.gtfs.loader;
+
+import com.conveyal.gtfs.model.Entity;
+import com.conveyal.gtfs.storage.StorageException;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonSyntaxException;
+import gnu.trove.set.TIntSet;
+import gnu.trove.set.hash.TIntHashSet;
+import org.apache.commons.dbutils.DbUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static com.conveyal.gtfs.loader.JdbcGtfsLoader.INSERT_BATCH_SIZE;
+
+/**
+ * This wraps a single database table and provides methods to modify GTFS entities.
+ */
+public class JdbcTableWriter implements TableWriter {
+    private static final Logger LOG = LoggerFactory.getLogger(JdbcTableWriter.class);
+    private final DataSource dataSource;
+    private final Table specTable;
+    private final String tablePrefix;
+    private static Gson gson = new GsonBuilder().create();
+
+    /**
+     * Enum containing available methods for updating in SQL.
+     */
+    private enum SqlMethod {
+        DELETE, UPDATE, CREATE
+    }
+
+    public JdbcTableWriter (Table specTable, DataSource dataSource, String tablePrefix) {
+        this.tablePrefix = tablePrefix;
+        this.dataSource = dataSource;
+        this.specTable = specTable;
+    }
+
+    private static JsonObject getJsonObject(String json) {
+        JsonElement jsonElement;
+        try {
+            jsonElement = gson.fromJson(json, JsonElement.class);
+        } catch (JsonSyntaxException e) {
+            LOG.error("Bad JSON syntax", e);
+            throw e;
+        }
+        return jsonElement.getAsJsonObject();
+    }
+
+    @Override
+    public String create(String json) throws SQLException {
+        return update(null, json);
+    }
+
+    @Override
+    public String update(Integer id, String json) throws SQLException {
+        final boolean isCreating = id == null;
+        JsonObject jsonObject = getJsonObject(json);
+        String tableName = String.join(".", tablePrefix, specTable.name);
+        // Parse the fields/values into a Field -> String map (drops ALL unknown fields)
```

Comment:
We should elaborate on what an "unknown field" is and probably mention this in the Javadoc for the method.